mediapipe/tasks/ios/genai/inference/sources/LlmInference.swift

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import LlmInferenceEngineC
+import LlmTaskRunner
+
+/// A MediaPipe task that performs inference using a given Large Language Model.
+///
+/// Note: Inherits from `NSObject` for Objective C interoperability.
+@objc(MPPLlmInference) public final class LlmInference: NSObject {
+  private static let numberOfDecodeStepsPerSync = 3
+  private static let sequenceBatchSize = 0
+
+  private let llmTaskRunner: LlmTaskRunner
+
+  /// Creates a new instance of `LlmInference` with the given options.
+  ///
+  /// - Parameters:
+  ///   - options: The options of type `LlmInference.Options` to use for configuring the
+  /// `LlmInference`.
+  @objc public init(options: Options) {
+    let modelPath = strdup(options.modelPath)
```

Comment:
Question: do we need to strdup here if we free the memory once we exit the method? The same behavior would be achieved by just passing the exiting parameter.

Reply:
I get a Swift warning if I don't do this. Logically yes. That was my original implementation.  Something along the lines of  "memory may be freed if you pass in this to the method." I did this to remove it. So users don't see any warnings when they use the library. But it didn't make sense to me either.